pkg/server/cache/entrycache/fullcache_ds.go

```diff
@@ -144,15 +148,23 @@ func (it *agentIteratorDS) fetchAgents(ctx context.Context) ([]Agent, error) {
 		return nil, err
 	}
 
-	agents := make([]Agent, 0, len(resp.Selectors))
+	// Aggregate selectors by agent ID. This is a defensive measure in case a
+	// datastore bug causes it to return agent selectors split across multiple
+	// response records.
+	agentSelectors := make(map[spiffeid.ID][]*common.Selector, len(resp.Selectors))
```

Comment:
I'm sure I must missing something, but is there a restriction to DB level to no allow duplicated spiffeIDs?